esp/services/ws_topology/ws_topologyService.cpp

```diff
@@ -236,6 +236,16 @@ bool CWsTopologyEx::onTpLogFile(IEspContext &context,IEspTpLogFileRequest  &req,
     return true;
 }
 
+void CWsTopologyEx::validateFilePath(const char *file, const char *fileType, const char *compType, const char *compName)
+{
+    StringBuffer actualPath;
+    splitUNCFilename(file, nullptr, &actualPath, nullptr, nullptr);
+    if (containsRelPaths(actualPath)) //Detect a path like: /home/lexis/runtime/var/log/HPCCSystems/myesp/../../../
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file path %s", actualPath.str());
+    if (!validateConfigurationDirectory(nullptr, fileType, compType, compName, actualPath))
```

Comment:
Potential performance issues because this will reread environment.xml each time it is called?  (getOSSdirTree() is not cached.)

Reply:
Do you know why getOSSdirTree() is not cached?

Reply:
Because no one has implemented it.  I noticed, and nearly add caching when I was modifying code in that area.  It needs to be a time limited cache - and should be a separate PR.

Reply:
Created HPCC-26316